Index: include/apply_submit.php
===================================================================
--- include/apply_submit.php	(revision 210)
+++ include/apply_submit.php	(working copy)
@@ -134,7 +134,7 @@
 //submits the application, doing all the checks and then generating PDFs and updating database along the way
 // if do_submit is false, then the PDFs will be generated but they want be added to the database
 // this is used for viewing the PDF prior to submission
-// returns: error message on fail, or PDF array on success (recommendations not in array)
+// returns: error message on fail, or PDF array on success (recommendations not included in array)
 function submitApplication($user_id, $application_id, $do_submit = true) {
 	$user_id = escape($user_id);
 	$application_id = escape($application_id);
@@ -157,6 +157,30 @@
 		return "application cannot be submitted at this time";
 	}
 	
+	//verify the user has not exceeded the limit category for the club
+	$result = mysql_query("SELECT limitcat FROM clubs WHERE id = '" . $checkResult[1] . "'");
+	if($row = mysql_fetch_array($result)) {
+		if(!is_null($row[0])) {
+			$limitcat = escape($row[0]);
+			$result = mysql_query("SELECT lim FROM club_limits WHERE limitcat = '$limitcat'");
+			
+			if($row = mysql_fetch_array($result)) {
+				$limit = escape($row[0]);
+				
+				//get the number of clubs in this limitcat that the user has applied to
+				$result = mysql_query("SELECT COUNT(applications.id) FROM applications, clubs WHERE applications.user_id = '$user_id' AND applications.submitted != '' AND applications.club_id = clubs.id AND clubs.limitcat = '$limitcat'");
+				$row = mysql_fetch_array($result);
+				$num_applied = $row[0];
+				
+				if($num_applied >= $limit) {
+					return "user applied to too many clubs in the limiting category, " . $limitcat;
+				}
+			}
+		}
+	} else {
+		return "internal error, club not found (2)";
+	}
+	
 	//verify that enough peer recommendations have been inputted; grab the filenames while we're at it
 	$result = mysql_query("SELECT num_recommend FROM clubs WHERE id = '" . $checkResult[1] . "'");
 	$recommendResult = mysql_query("SELECT filename FROM recommendations WHERE user_id = '$user_id' AND status = '1'");
Index: include/common.php
===================================================================
--- include/common.php	(revision 210)
+++ include/common.php	(working copy)
@@ -695,11 +695,11 @@
 //returns array of (club id, club name, club description, user's application id)
 function getUserClubsApplied($user_id) {
 	$user_id = escape($user_id);
-	$result = mysql_query("SELECT applications.club_id, clubs.name, clubs.description, applications.id FROM applications, clubs WHERE applications.user_id='$user_id' AND applications.club_id = clubs.id AND applications.club_id != '0' ORDER BY clubs.name");
+	$result = mysql_query("SELECT applications.club_id, clubs.name, clubs.description, applications.id, clubs.limitcat FROM applications, clubs WHERE applications.user_id='$user_id' AND applications.club_id = clubs.id AND applications.club_id != '0' ORDER BY clubs.name");
 	
 	$clubs = array();
 	while($row = mysql_fetch_array($result)) {
-		array_push($clubs, array($row[0], $row[1], $row[2], $row[3]));
+		array_push($clubs, array($row[0], $row[1], $row[2], $row[3], $row[4]));
 	}
 	
 	return $clubs;
@@ -719,13 +719,20 @@
 	}
 }
 
-//returns array (club name, club description, open_time, close_time, num_recommendations)
+//returns array (club name, club description, open_time, close_time, num_recommendations, limitcat, limit)
 function clubInfo($club_id) {
 	$club_id = escape($club_id);
-	$result = mysql_query("SELECT name, description, open_time, close_time, num_recommend FROM clubs WHERE id='$club_id'");
+	$result = mysql_query("SELECT name, description, open_time, close_time, num_recommend, limitcat FROM clubs WHERE id='$club_id'");
 	
 	if($row = mysql_fetch_array($result)) {
-		return array($row[0], $row[1], clubTimeString($row[2]), clubTimeString($row[3]), $row[4]);
+		$limitResult = mysql_query("SELECT lim FROM club_limits WHERE limitcat = '" . escape($row[5]) . "'");
+		if($limitRow = mysql_fetch_array($limitResult)) {
+			$limit = $limitRow[0];
+		} else {
+			$limit = FALSE;
+		}
+		
+		return array($row[0], $row[1], clubTimeString($row[2]), clubTimeString($row[3]), $row[4], $row[5], $limit);
 	} else {
 		return array("Unknown", "Club could not be found");
 	}
Index: root/man_clubs.php
===================================================================
--- root/man_clubs.php	(revision 210)
+++ root/man_clubs.php	(working copy)
@@ -27,7 +27,7 @@
 			}
 			
 			if(!$limitFail) {
-				mysql_query("INSERT INTO clubs (name, description, view_time, open_time, close_time) VALUES ('$name', '$description', '0', '0', '0')");
+				mysql_query("INSERT INTO clubs (name, description, view_time, open_time, close_time, limitcat) VALUES ('$name', '$description', '0', '0', '0', '')");
 				$message = "Club added successfully! Click <a href=\"man_clubs.php\">here</a> to continue.";
 			}
 		} else if($action == 'delete') {
@@ -37,14 +37,16 @@
 		} else if($action == 'update') {
 			$club_id = escape($_REQUEST['id']);
 			$description = escape($_REQUEST['description']);
+			$limitcat = escape($_REQUEST['limitcat']);
 			
-			mysql_query("UPDATE clubs SET description='$description' WHERE id='$club_id'");
+			mysql_query("UPDATE clubs SET description='$description', limitcat='$limitcat' WHERE id='$club_id'");
 			$message = "Club updated successfully! Click <a href=\"man_clubs.php\">here</a> to continue.";
 		}
 	}
 	
-	$result = mysql_query("SELECT id,name,description FROM clubs");
-	get_page_advanced("man_clubs", "root", array('message' => $message, 'clubsResult' => $result));
+	$result = mysql_query("SELECT id,name,description,limitcat FROM clubs");
+	$clubLimit = mysql_query("SELECT limitcat, lim FROM club_limits");
+	get_page_advanced("man_clubs", "root", array('message' => $message, 'clubsResult' => $result, 'clubLimit' => $clubLimit));
 } else {
 	header('Location: index.php');
 }
Index: root/man_limit.php
===================================================================
--- root/man_limit.php	(revision 0)
+++ root/man_limit.php	(revision 0)
@@ -0,0 +1,35 @@
+<?php
+include("../config.php");
+include("../include/common.php");
+include("../include/db_connect.php");
+include("../include/session.php");
+
+if(isset($_SESSION['root'])) {
+	$message = "";
+	
+	if(isset($_REQUEST['action'])) {
+		$action = $_REQUEST['action'];
+		
+		if($action == 'add') {
+			$limitcat = escape($_REQUEST['limitcat']);
+			$limit = escape($_REQUEST['limit']);
+			mysql_query("INSERT INTO club_limits (limitcat, lim) VALUES ('$limitcat', '$limit')");
+		} else if($action == 'delete') {
+			$limitcat = escape($_REQUEST['limitcat']);
+			mysql_query("UPDATE clubs SET limitcat = '' WHERE limitcat = '$limitcat'");
+			mysql_query("DELETE FROM club_limits WHERE limitcat = '$limitcat'");
+		} else if($action == 'update') {
+			$limitcat = escape($_REQUEST['limitcat']);
+			$limit = escape($_REQUEST['limit']);
+			mysql_query("UPDATE basecat SET limit = '$limit' WHERE limitcat = '$limitcat'");
+			mysql_query("UPDATE club_limits SET lim = '$limit' WHERE limitcat = '$limitcat'");
+		}
+	}
+
+	$result = mysql_query("SELECT limitcat, lim FROM club_limits");
+	
+	get_page_advanced("man_limit", "root", array('result' => $result));
+} else {
+	header('Location: index.php');
+}
+?>
Index: config_.php
===================================================================
--- config_.php	(revision 210)
+++ config_.php	(working copy)
@@ -66,8 +66,8 @@
 $config['root_page_display'] = array('index', 'index.php?action=logout&ex=');
 $config['root_page_display_names'] = array('Home', 'Logout');
 
-$config['root_side_display'] = array('index', 'man_pages', 'man_admins', 'man_cat', 'man_clubs', 'userlist', 'rm_peer', 'check_pdf', 'check_nohome', 'check_mismatch', 'statistics', 'statistics_club', 'man_config', 'full_clean', 'dbwipe', 'backup');
-$config['root_side_display_names'] = array('Root Home', 'Manage pages', 'Manage admins', 'Manage categories', 'Manage clubs', 'User list', 'Remove recommendations', 'Check extra PDFs', 'Check questions without a home', 'Check mismatched applications', 'Statistics', 'Club Statistics', 'Edit configuration', 'Full database cleaner', 'Database wipe', 'Backup');
+$config['root_side_display'] = array('index', 'man_pages', 'man_admins', 'man_cat', 'man_clubs', 'man_limit', 'userlist', 'rm_peer', 'check_pdf', 'check_nohome', 'check_mismatch', 'statistics', 'statistics_club', 'man_config', 'full_clean', 'dbwipe', 'backup');
+$config['root_side_display_names'] = array('Root Home', 'Manage pages', 'Manage admins', 'Manage categories', 'Manage clubs', 'Manage Restrictions', 'User list', 'Remove recommendations', 'Check extra PDFs', 'Check questions without a home', 'Check mismatched applications', 'Statistics', 'Club Statistics', 'Edit configuration', 'Full database cleaner', 'Database wipe', 'Backup');
 
 $config['time_dateformat'] = 'D, d M Y H:i:s'; //format used to display the current time
 $config['club_dateformat'] = 'D, d M Y H:i:s'; //format used to display deadlines and opening times for club supplements
Index: astyle/one/header.php
===================================================================
--- astyle/one/header.php	(revision 210)
+++ astyle/one/header.php	(working copy)
@@ -48,6 +48,7 @@
 	</div>
 	
 	<div id="navbar"><table width=100% background-color="red"><tr><td><a id="clockbox"><?= $timeString ?></a></td><td align="right">
+	<a href="http://www.one-app.org/OneApp_Manual.pdf">Manual</a>
 <?
 for($i = 0; $i < count($page_display); $i++) {
 	echo '<a href=' . $page_display[$i] . '.php>' . $page_display_names[$i] . '</a> ';
Index: application/clubs.php
===================================================================
--- application/clubs.php	(revision 210)
+++ application/clubs.php	(working copy)
@@ -23,7 +23,10 @@
 		$clubClose[$item[0]] = $clubInfo[3];
 	}
 
-	get_page_advanced("clubs", "apply", array("clubsApplied" => $clubsApplied, 'clubStates' => $clubStates, 'clubStart' => $clubStart, 'clubClose' => $clubClose));
+	
+	$clubLimit = mysql_query("SELECT limitcat, lim FROM club_limits");
+	
+	get_page_advanced("clubs", "apply", array("clubsApplied" => $clubsApplied, 'clubStates' => $clubStates, 'clubStart' => $clubStart, 'clubClose' => $clubClose, 'clubLimit'=> $clubLimit));
 } else {
 	get_page_advanced("message", "apply", array("title" => "Not Logged In", "message" => "You cannot access the application because you are not logged in. Please <a href=\"../login.php\">login first</a>."));
 }
Index: page/root/statistics_club.php
===================================================================
--- page/root/statistics_club.php	(revision 210)
+++ page/root/statistics_club.php	(working copy)
@@ -16,11 +16,11 @@
 ?>
 	<tr>
 	<td><p class="admin_table_entry"><?= $clubTuple[0] ?></p></td>
-	<td align="center"><p class="admin_table_entry"><?= $clubTuple[1] ?> / <?= $clubTuple[2] ?></p></td>
-	<td>
+	<td align="center"><p class="admin_table_entry"><?= $clubTuple[1] ?> / <?= $clubTuple[2]-1 ?></p></td>
+	<td align="center">
 		<img src="../include/ratingbox.php?rating=<?= $clubTuple[1] * 100 / $clubTuple[2] ?>" width=50%>
 	</td>
-	<td><p class="admin_table_entry"><a href="statistics_club.php?club_id=<?= $club_id ?>&club_name=<?= $clubTuple[0] ?>">Details</a></p></td>
+	<td align="center"><p class="admin_table_entry"><a href="statistics_club.php?club_id=<?= $club_id ?>&club_name=<?= $clubTuple[0] ?>">Details</a></p></td>
 	</tr>
 <? } ?>
 
Index: page/root/man_clubs.php
===================================================================
--- page/root/man_clubs.php	(revision 210)
+++ page/root/man_clubs.php	(working copy)
@@ -24,7 +24,6 @@
 	} else {
 		$class_string = "";
 	}
-	
 	$band_class = ($counter + 1) % 2 + 1;
 ?>
 	<tr><table <?=$class_string?> width=100% cellspacing=0>
@@ -33,6 +32,21 @@
 	<input type="hidden" name="id" value="<?= $row['id'] ?>">
 	<tr class="band<?=$band_class?>"><td width=20%><p style="font-weight:bold">Club ID:</p></td><td><p><?= $row['id'] ?></p></td></tr>
 	<tr class="band<?=$band_class?>"><td><p style="font-weight:bold">Club Name:</p></td><td><p><?= $row['name'] ?></p></td></tr>
+	<tr class="band<?=$band_class?>"><td><p style="font-weight:bold">Club Category:</p></td><td>
+	
+	<select name="limitcat">
+		<option value="<?= $row['limitcat'] ?>" /><?= $row['limitcat'] ?></option>
+		<option value="">-------</option>
+		<?
+		$result = mysql_query("SELECT limitcat, lim FROM club_limits");
+		 while($item = mysql_fetch_row($result)) {
+				echo "<option value=\"" . $item[0] . "\" />" . $item[0] . "</option>";	
+		 }
+		?>
+	</select>
+	</td></tr>
+	
+	
 	<tr class="band<?=$band_class?>" align="center"><td colspan="2"><p style="font-weight:bold" align="left">Description:</p><textarea name="description" style="width:95%;height:100;resize:none" class="band<?=$band_class?>"><?= $row['description'] ?></textarea></td></tr>
 	<tr class="band<?=$band_class?>"><td></td><td align="right"><input type="submit" name="action" value="update"><input type="submit" name="action" value="delete"><br></td></tr>
 	</form>
Index: page/root/man_limit.php
===================================================================
--- page/root/man_limit.php	(revision 0)
+++ page/root/man_limit.php	(revision 0)
@@ -0,0 +1,34 @@
+<h1>Limit manager</h1>
+
+<p>This is for the limit extension, allowing you to edit limits. Limit category name identifies the limit category, and should be the same as the value in the club manager. The limit will determine the maximum submissions in the specified limit category per user. Do not set a category for a club if the club should not be limited (there is no way to create a category without a limit).</p>
+
+<form action="man_limit.php?action=add" method="post">
+<table>
+<tr><td><p>Limit category</p></td><td><input type="text" name="limitcat"></td></tr>
+<tr><td><p>Limit</p></td><td><input type="text" name="limit"></td></tr>
+<tr><td colspan="2"><input type="submit" value="Add category"></td></tr>
+</table>
+</form>
+		
+<table width=100% cellspacing=0>
+<tr>
+	<th><p class="admin_table_header">Limit category</p></th>
+	<th><p class="admin_table_header">Limit</p></th>
+	<th><p class="admin_table_header">Update</p></th>
+	<th><p class="admin_table_header">Delete</p></th>
+</tr>
+
+<?
+while($row = mysql_fetch_array($result)) {
+?>
+	<form method="post" action="man_limit.php"><tr align="center">
+	<td><input type="text" name="limitcat" value="<?= $row[0] ?>" style="width:100%"></td>
+	<td><input type="text" name="limit" value="<?= $row[1] ?>" style="width:100%"></td>
+	<td><input type="submit" name="action" value="update"></td>
+	<td><input type="submit" name="action" value="delete"></td>
+	</tr></form>
+<?
+}
+?>
+
+</table>
Index: page/apply/clubs.php
===================================================================
--- page/apply/clubs.php	(revision 210)
+++ page/apply/clubs.php	(working copy)
@@ -8,9 +8,29 @@
 
 <br />
 
+<? 
+if(count($clubLimit) != 0) {
+?>
+	<p>The following restrictions have been applied by your administrator<p>
+	<table class="borderon">
+		<tr>
+			<th width="20%"><p class="admin_table_header">Category</p></th>
+			<th><p class="admin_table_header">Number of Applications Allowed</p></th>
+		</tr>
+<?
+	while($item = mysql_fetch_array($clubLimit)) {
+		echo "<tr align=\"center\"><td><p><b>" . $item[0] . "</b></p></td><td><p>" . $item[1] . "</p></td></tr>";
+	}
+}
+?>
+</table>
+
+<br />
+
 <table width=100% class="borderon">
 <tr>
        <th align="left"><p class="admin_table_header">Club Name</p></th>
+       <th><p class="admin_table_header">Category</p></th>
        <th><p class="admin_table_header">Status</p></th>
        <th><p class="admin_table_header">Open time</p></th>
        <th><p class="admin_table_header">Close time</p></th>
@@ -22,15 +42,15 @@
 	$club_id = $item[0];
 	$club_name = $item[1];
 	$app_id = $item[3];
+	$club_cat = $item[4];
 	
 	echo "<tr align=\"center\">";
-	echo "<td width=40% align=\"left\"><p>";
-	echo "<a href=\"clubDetail.php?app_id=$app_id&club_id=$club_id\">";
-	echo $club_name . "</p></a></td>";
-	echo "<td width=20%><p>" . $clubStates[$club_id] . "</p></td>";
-	echo "<td width=20%><p>" . $clubStart[$club_id] . "</p></td>";
-	echo "<td width=20%><p>" . $clubClose[$club_id] . "</p></td>";
-	echo "<td width=20%><p>";
+	echo "<td align=\"left\"><p><a href=\"clubDetail.php?app_id=$app_id&club_id=$club_id\">" . $club_name . "</p></a></td>";
+	echo "<td><p>" . $club_cat . "</p></td>";
+	echo "<td><p>" . $clubStates[$club_id] . "</p></td>";
+	echo "<td><p>" . $clubStart[$club_id] . "</p></td>";
+	echo "<td><p>" . $clubClose[$club_id] . "</p></td>";
+	echo "<td><p>";
 	if($clubStates[$club_id] == "<font color=\"blue\">Started</font>") {
 		echo "<a href=\"submit.php?app_id=" . $item[3] . "&club_id=" . $item[0] . "\">Submit</a>";
 	} else {
Index: page/affclubs_info.php
===================================================================
--- page/affclubs_info.php	(revision 210)
+++ page/affclubs_info.php	(working copy)
@@ -3,6 +3,10 @@
 <p><?= htmlentities($clubInfo[1]) ?></p>
 
 <p>Submission window: between <?= $clubInfo[2] ?> and <?= $clubInfo[3] ?><br />
-Number of recommendations required: <?= $clubInfo[4] ?></p>
+Number of recommendations required: <?= $clubInfo[4] ?>
+<? if($clubInfo[5] != '' && $clubInfo[6] !== FALSE) { ?>
+<br />Limitations: category is <?= $clubInfo[5] ?> (category limit: <?= $clubInfo[6] ?> applications)
+<? } ?>
+</p>
 
 <p><a href="affclubs.php">Back</a> to affiliated clubs list.</p>
